### ISSUE TITLE
Ldflags and "linker"

### ISSUE THREAD

**Issue by vron** [gopherjs/gopherjs#325]
Hi,

I often use the -X importpath.name=value functionality to implement versions of my applications (e.g. pulling the git hash as part of my build process). I think this, or something similar, would be a useful addition to gopherjs. Any thoughs?

Would a change/addition like this be accepted? Would anyone mind pointing me in the correct direction in the source?

**Comment by neelance** [comment]
Yes, that feature would be welcome. :-) The first implementation that comes to my mind would be to fake entries in `InitOrder` here: https://github.com/gopherjs/gopherjs/blob/master/compiler/package.go#L247

**Comment by neelance** [comment]
@vron Are you looking into it? If you get stuck, feel free to ask me on the Gophers Slack. I could also look into this myself, I just want to make sure that we're not working on the feature at the same time.

**Comment by vron** [comment]
@neelance I did look into it a bit, but since I am not fluent in the gopherjs code it became slighty more compicated than I hoped. I have solved my direct need in other ways (updating source). I.e. I am not currently looking at this, no.

**Comment by shurcooL** [comment]
+1, I'm interested in this functionality too. Maybe I'll try to look into it soon.

**Comment by shurcooL** [comment]
What should the API for this look like?

Does GopherJS have a concept of a distinct compiler and **linker** @neelance?

Would we want to support any other of go linker flags?

**Comment by shurcooL** [comment]
This issue will not be automatically/immediately resolved by my WIP implementation of #388. It will be possible, but there's still custom work needed to support this for JS environment.

That's why I'll remove the "cmd/go would not have this issue" label from this issue. We can implement it after #388 is resolved.

**Comment by pjebs** [comment]
@dmitshur @neelance Will this be merged?

**Comment by nevkontakte** [comment]
@pjebs I'm afraid there is no current implementation for this feature, although I agree it'd be nice to have.
